cups/file.c

```diff
@@ -841,6 +841,142 @@ cupsFileGetConf(cups_file_t *fp,	/* I  - CUPS file */
   return (NULL);
 }
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get a line and Comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(
+    cups_file_t *fp,	    /* I  - CUPS file */
+    char        *buf,	    /* O  - String buffer */
+    size_t      buflen,	    /* I  - Size of string buffer */
+    char        **value,    /* O  - Pointer to value */
+    int         *linenum,   /* IO - Current line number */
+    char        *comment,   /* I  - Comment */
+    size_t      commentlen) /* I  - Size of comment buffer */
+{
+  char	*ptr;				/* Pointer into line */
+
+/*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p, comment=%p, commentlen=" CUPS_LLFMT ", comment_linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum, (void *)comment, CUPS_LLCAST commentlen));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value || !comment || commentlen<2)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next non-comment line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+   /*
+    * Strip any comments...
+    */
+
+    if ((ptr = strchr(buf, '#')) != NULL)
+    {
```

Comment:
If we go with passing comments via ```line```, we can detect the comment, then remove the leading and trailing whitespaces and return ```buf```.

Reply:
@zdohnal I made the changes as you requested. Now I 'm using line variable to get comments. Please review them.